src/conn/conn_dhandle.c

```diff
@@ -134,7 +134,17 @@ __conn_dhandle_mark_dead(WT_SESSION_IMPL *session)
 	 * eviction and drain any blocks already queued.
 	 */
 	WT_RET(__wt_evict_file_exclusive_on(session, &evict_reset));
+
+	/* The handle is dead. */
 	F_SET(session->dhandle, WT_DHANDLE_DEAD);
+
+	/*
+	 * Reset the tree's eviction priority, and the tree is evictable by
+	 * definition.
+	 */
+	__wt_evict_priority_clear(session);
+	F_CLR(S2BT(session), WT_BTREE_NO_EVICTION);
+
 	if (evict_reset)
 		__wt_evict_file_exclusive_off(session);
```

Comment:
@michaelcahill 
> but maybe we just make the call to __wt_evict_file_exclusive_off here unconditional on evict_reset, along with a comment.

This code was reshuffled as part of your changes to ef9927e, where you moved the `__wt_evict_file_exclusive_on/off` calls, so the two actions are no longer adjacent.